src/TraceEvent/Computers/TraceManagedProcess.cs

```diff
@@ -290,14 +290,27 @@ internal static void SetupCallbacks(TraceEventDispatcher source)
                 var clrRundown = new ClrRundownTraceEventParser(source);
                 clrRundown.RuntimeStart += doAtRuntimeStart;
                 source.Clr.RuntimeStart += doAtRuntimeStart;
+                Dictionary<TraceProcess, TraceLoadedDotNetRuntime> processRuntimes = null;
 
                 CircularBuffer<ThreadWorkSpan> RecentThreadSwitches = new CircularBuffer<ThreadWorkSpan>(1000);
                 source.Kernel.ThreadCSwitch += delegate (CSwitchTraceData data)
                 {
```

Comment:
I know that CSwitches are enormously verbose, but is there a reason this cache is not applied more broadly?  If you look above in the file you can see how the lookup is done.  Perhaps using the ProcessIndex could offer a faster lookup.